src/praisonai-bot/praisonai_bot/scheduler/executor.py

```
@@ -749,6 +802,116 @@ def _resolve_condition(self, job: "ScheduleJob") -> Optional["JobConditionProtoc
+        if hasattr(store, "get_state") and hasattr(store, "set_state"):
+            return store
+        return None
+
+    def _get_job_state(self, job: "ScheduleJob") -> Dict[str, Any]:
+        """Load the job's persisted scratchpad (``{}`` when unsupported/empty)."""
+        store = self._state_store()
+        if store is None:
+            return {}
+        try:
+            state = store.get_state(job.id)
+        except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)